File: sdk/typescript/src/inputStep.ts
import { DependsOn, Field } from './types'

export interface InputStep {
    input: string;
    fields: Field[];
    allow_dependency_failure?: boolean;
    branches?: string[] | string;
    depends_on?: Array<DependsOn | string> | null | string;
    id?: string;
    identifier?: string;
    if?: string;
    key?: string;
    label?: string;
    name?: string;
    prompt?: string;
    blocked_state: 'passed' | 'failed' | 'running'
}

File: sdk/csharp/src/Buildkite.Sdk/InputStep.cs
using System.Collections.Generic;

namespace Buildkite.Sdk
{
    public class InputStep : PipelineStep
    {
        public string? Input { get; set; }
        public string? Prompt { get; set; }
        public List<Field>? Fields { get; set; }
        public BlockedState? BlockedState { get; set; }

        public InputStep()
        {
        }

        public InputStep(string input)
        {
            Input = input;
        }

        public InputStep(string label, string input)
        {
            Label = label;
            Input = input;
        }
    }
}